shared/src/main/scala/scala/xml/Utility.scala

```diff
@@ -46,17 +46,28 @@ object Utility extends AnyRef with parsing.TokenTests {
    */
   def trim(x: Node): Node = x match {
     case Elem(pre, lab, md, scp, child@_*) =>
-      val children = child flatMap trimProper
+      val children = combineAdjacentTextNodes(child:_*) flatMap trimProper
       Elem(pre, lab, md, scp, children.isEmpty, children: _*)
   }
 
+  private def combineAdjacentTextNodes(children: Node*): Seq[Node] = {
+    children.foldLeft(Seq.empty[Node]) { (acc, n) =>
+      (acc.lastOption, n) match {
+        case (Some(Text(l)), Text(r)) => {
+          acc.dropRight(1) :+ Text(l + r)
+        }
+        case _ => acc :+ n
+      }
```

Comment:
Right, `foldRight` I always use foldLeft so it never comes to mind. That would work, 

```
import scala.xml._
val node = <A><One>1</One>{Text("Hello ")}{Text("World")}{Text(" I like pie.")}<Two>2</Two><Three><Nope>no</Nope></Three></A>

node.child.foldRight(Seq.empty[Node]) {
	case (Text(left), Text(right) +: accMinusLast) => Text(left + right) +: accMinusLast
	case (n, acc) => n +: acc 
}

// Seq[scala.xml.Node] = List(<One>1</One>, Hello World I like pie., <Two>2</Two>, <Three><Nope>no</Nope></Three>)
```

Looks like it would. I wasn't aware you can pattern match the last element in a list but I think that code is more elegant so I can update the PR with that and the tests will inform us if we're getting the same result (answer is probably yes)